include/seqan3/io/alignment_file/format_sam.hpp

```diff
@@ -556,17 +556,19 @@ class alignment_file_input_format_REMOVEME<format_sam>
      */
     void transfer_soft_clipping_to(std::vector<cigar> const & cigar_vector, int32_t & sc_begin, int32_t & sc_end) const
     {
+        auto v_size = cigar_vector.size();
```

Comment:
well it is much better. Since it is a fix. We merge it right now. I will make another PR to clean the code a little up.